### ISSUE TITLE
quic-client fuzzer crash: crypto/packet.c:135: OpenSSL internal error: Assertion failed: buf != NULL && len > 0

### ISSUE THREAD

**Issue by kroeckx** [openssl/openssl#22699]
I'm getting: crypto/packet.c:135: OpenSSL internal error: Assertion failed: buf != NULL && len > 0
from this file: 
[crash-1aa7e94f101c9b954617cdd257f4e21a548ae71a.gz](https://github.com/openssl/openssl/files/13324204/crash-1aa7e94f101c9b954617cdd257f4e21a548ae71a.gz)

**Comment by mattcaswell** [comment]
Hmmm. Interestingly I get a different failure from this crash file:

````
$ ./quic-client-test ~/Downloads/crash-1aa7e94f101c9b954617cdd257f4e21a548ae71a 
# /home/matt/Downloads/crash-1aa7e94f101c9b954617cdd257f4e21a548ae71a
ssl/quic/quic_txp.c:297:41: runtime error: applying zero offset to null pointer
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ssl/quic/quic_txp.c:297:41 in 
````

**Comment by mattcaswell** [comment]
On the assumption that the failure you see, and the failure I see have the same underlying cause the fix is in #22715
